python/foglamp/common/configuration_manager.py

```diff
@@ -30,6 +31,56 @@
                               'URL', 'enumeration'])
 
 
+class CachedConfiguration(object):
+    """Cached Configuration Manager"""
+
+    MAX_CACHE_SIZE = 10
+
+    def __init__(self):
+        """
+        cache: value dictionary as per category_name
+        max_cache_size: Hold the 10 recently requested categories in the cache
+        hit: number of times an item is read from the cache
+        miss: number of times an item was not found in the cache and a read of the storage layer was required
+        """
+        self.cache = {}
+        self.max_cache_size = self.MAX_CACHE_SIZE
+        self.hit = 0
+        self.miss = 0
+
+    def __contains__(self, category_name):
+        """Returns True or False depending on whether or not the key is in the cache
+        and update the hit and data_accessed value only If True"""
+        if category_name in self.cache:
+            current_hit = self.cache[category_name]['hit']
+            self.cache[category_name].update({'date_accessed': datetime.datetime.now(), 'hit': current_hit + 1})
+            return True
+        return False
+
+    def update(self, category_name, category):
+        """Update the cache dictionary and remove the oldest item"""
+        if category_name not in self.cache and len(self.cache) >= self.max_cache_size:
+            self.remove_oldest()
+
+        self.cache[category_name] = {'date_accessed': datetime.datetime.now(), 'value': category, 'hit': self.hit, 'miss': self.miss}
```

Comment:
not self.hit self.miss but corresponding increments to category_name only, as you did in  `__contains__` def. for class level overall stats you can use these with self, but please increment by 1  these when these are hit or miss

Reply:
use update and update only date_accessed and value.

> hit and miss will be handled by ` __contains__`